DESIGN.md

```diff
@@ -726,6 +728,10 @@ operation and return it as part of the second read. The process would not lose
 data, but the runtime would need to perform an extra copy or return the caller a
 different buffer than the one it submitted.
 
+
+The proposed API is not vulnerable to this issue as resources track their
+operations, preventing them from being dropped as long as the resource is open.
```

Comment:
I don't think that's the reason this problem wouldn't occur? If someone dropped a call to `consume` in a select, what would happen?

Reply:
consume is a future, but it is not an **operation** future.

Reply:
Sure, but dropping the `consume` (just like dropping the `read`) might still lead to bytes being lost, no?